Foundation/URLSession/URLSession.swift

```diff
@@ -192,6 +192,7 @@ public let NSURLSessionTransferSizeUnknown: Int64 = -1
 open class URLSession : NSObject {
     fileprivate let _configuration: _Configuration
     fileprivate let multiHandle: _MultiHandle
+    fileprivate let taskIdentifierLock = NSLock()
```

Comment:
also: the `sessionDescription` and `delegate` properties are also mutable and accessible from the outside. I suppose the should be locked too.